components/keys/src/types.rs

```diff
@@ -274,4 +283,13 @@ mod tests {
         assert_eq!(false, eq(b"axcdefghijk87654321", b"abcdefghijk"));
         assert_eq!(false, eq(b"abcdeffhijk87654321", b"abcdefghijk"));
     }
+
+    #[test]
+    fn test_is_encoded_from() {
+        for raw_len in 0..=255 {
+            let raw: Vec<u8> = (0..raw_len).collect();
+            let encoded = Key::from_raw(&raw);
+            assert!(encoded.is_encoded_from(&raw));
```

Comment:
Should test negative cases too.

Reply:
Added. PTAL again

Tests are deliberately repeated in tikv_util::codec and keys, in case we change the implementation in keys to components/codec.